api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxNewCohortQueries.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.eptsreports.reporting.library.cohorts;
+
+import java.util.Date;
+
+import org.openmrs.Location;
+import org.openmrs.module.eptsreports.metadata.HivMetadata;
+import org.openmrs.module.eptsreports.metadata.TbMetadata;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.CompositionCohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.SqlCohortDefinition;
+import org.openmrs.module.reporting.definition.library.DocumentedDefinition;
+import org.openmrs.module.reporting.evaluation.parameter.Mapped;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.openmrs.module.reporting.evaluation.parameter.ParameterizableUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Defines all of the TxNew Cohort Definition instances we want to expose for EPTS
+ */
+@Component
+public class TxNewCohortQueries {
+	
+	@Autowired
+	private HivMetadata hivMetadata;
+	
+	@Autowired
+	private TbMetadata tbMetadata;
+	
+	// Looks for patients enrolled in ART program (program 2=SERVICO TARV -
+	// TRATAMENTO) before or on end date
+	@DocumentedDefinition(value = "inARTProgramDuringTimePeriod")
+	public CohortDefinition getPatientsinARTProgramDuringTimePeriod() {
+		SqlCohortDefinition inARTProgramDuringTimePeriod = new SqlCohortDefinition();
+		inARTProgramDuringTimePeriod.setName("inARTProgramDuringTimePeriod");
+		inARTProgramDuringTimePeriod
+		        .setQuery("select p.patient_id from patient p inner join patient_program pg on p.patient_id=pg.patient_id "
+		                + "where pg.voided=0 and p.voided=0 and pg.program_id= " + hivMetadata.getARTProgram().getProgramId()
+		                + " and pg.date_enrolled <= :onOrBefore and and pg.location_id=:location");
```

Comment:
Small typo here `:onOrBefore and and pg.location_id`

Reply:
Not really a typo because that depends on the name of the parameter - in this case the parameter is called `onOrBefore`. We could change it to be `endDate`, but it won't change the query.

Reply:
It would sound correct if it is enrolledOnOrBefore and mapped to endDate in respective compositions and indicators, like it is here https://github.com/openmrs/openmrs-module-reporting/blob/master/api/src/main/java/org/openmrs/module/reporting/cohort/definition/ProgramEnrollmentCohortDefinition.java#L39

Reply:
I was referring to the `and and` :sweat_smile:

Reply:
lol gotcha :+1: